src/org/jitsi/impl/neomedia/rtp/RTPEncodingImpl.java

```diff
@@ -168,6 +168,62 @@ public RTPEncodingImpl(
     }
 
     /**
+     * Applies frame boundaries heuristics to frames a and b, assuming a
+     * predates/is older than b.
+     *
+     * @param a the old {@link FrameDesc}.
+     * @param b the new {@link FrameDesc}
+     */
+    private static void applyFrameBoundsHeuristics(FrameDesc a, FrameDesc b)
+    {
+        int end = a.getEnd(), start = b.getStart();
+        if (end != -1 && start != -1)
+        {
+            // No need for heuristics.
+            return;
+        }
+
+        long tsDiff = (b.getTimestamp() - a.getTimestamp()) & 0xFFFFFFFFL;
+        if (tsDiff > (1L << 30) && tsDiff < (-(1L << 30) & 0xFFFFFFFFL))
+        {
+            // the distance (mod 32) between the two timestamps needs to be
+            // less than half the timestamp space.
+            return;
+        }
+
+        int min = b.getMinSeen(), max = a.getMaxSeen();
+        int snDiff = (max - min) & 0xFFFF;
+
+        if (start != -1 || end != -1)
+        {
+            // We can deal with sequence number distances (mod 16) that are
+            // less than 3.
+            if (snDiff < 3 || snDiff > (-3 & 0xFFFF))
+            {
+                if (end == -1)
+                {
+                    a.setEnd((max + 1) & 0xFFFF);
+                }
+
+                if (start == -1)
+                {
+                    b.setStart((min - 1) & 0xFFFF);
+                }
+            }
+        }
+        else
+        {
+            // We can deal with sequence number distances (mod 16) that are
+            // less than 4.
+            if (snDiff < 4 || snDiff > (-4 & 0xFFFF))
```

Comment:
We can apply this only when (start != -1 && end && -1 && diff == 3)

If either start or end is set, then there could be a 1-packet frame in-between. If diff is <3 there is probably something wrong.